test/widgets/settings_test.dart

```diff
@@ -1,35 +1,78 @@
 import 'package:checks/checks.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_checks/flutter_checks.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:legacy_checks/legacy_checks.dart';
 import 'package:zulip/model/settings.dart';
+import 'package:zulip/widgets/page.dart';
 import 'package:zulip/widgets/settings.dart';
+import 'package:zulip/widgets/store.dart';
 
 import '../flutter_checks.dart';
 import '../model/binding.dart';
 import '../model/store_checks.dart';
 import '../example_data.dart' as eg;
+import '../test_navigation.dart';
+import 'page_checks.dart';
 import 'test_app.dart';
 
 void main() {
   TestZulipBinding.ensureInitialized();
 
+  late TestNavigatorObserver testNavObserver;
+  late Route<dynamic>? lastPushedRoute;
+  late Route<dynamic>? lastPoppedRoute;
+
   Future<void> prepare(WidgetTester tester) async {
     addTearDown(testBinding.reset);
 
+    testNavObserver = TestNavigatorObserver()
+      ..onPushed = ((route, _) => lastPushedRoute = route)
+      ..onPopped = ((route, _) => lastPoppedRoute = route);
+    lastPushedRoute = null;
+    lastPoppedRoute = null;
+
     await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
     await tester.pumpWidget(TestZulipApp(
       accountId: eg.selfAccount.id,
+      navigatorObservers: [testNavObserver],
       child: SettingsPage()));
     await tester.pump();
     await tester.pump();
   }
 
-  group('ThemeSetting', () {
-    Finder findRadioListTileWithTitle(String title) => find.ancestor(
-      of: find.text(title),
-      matching: find.byType(RadioListTile<ThemeSetting?>));
+  void checkTileOnSettingsPage(WidgetTester tester, {
+    required String expectedTitle,
+    required String expectedSubtitle,
+  }) {
+    check(find.descendant(of: find.widgetWithText(ListTile, expectedTitle),
+      matching: find.text(expectedSubtitle))).findsOne();
+  }
+
+  Finder findRadioListTileWithTitle<T>(String title) => find.ancestor(
+    of: find.text(title),
+    matching: find.byType(RadioListTile<T>));
+
+  void checkRadioButtonAppearsChecked<T>(WidgetTester tester,
+      String title, bool expectedIsChecked, {String? subtitle}) {
+    check(tester.semantics.find(findRadioListTileWithTitle<T>(title)))
+      // `.legacyMatcher()` wrapper just because flutter_checks's
+      //   `matchesSemantics` doesn't yet have a `hasSelectedState` param
+      .legacyMatcher(matchesSemantics(
+        label: subtitle == null
+          ? title
+          : '$title\n$subtitle',
+        isFocusable: true,
+        hasEnabledState: true, isEnabled: true,
+        isInMutuallyExclusiveGroup: true,
+        hasCheckedState: true, isChecked: expectedIsChecked,
+        hasSelectedState: true,
+        hasTapAction: true,
+        hasFocusAction: true));
```

Comment:
Hmm, annoying we have to specify all these flags that seem like not the point of these tests. But I see that's how `matchesSemantics` behaves.

Reply:
… Oho: how about `containsSemantics`? Then I think this whole check statement becomes:
```dart
    check(tester.semantics.find(findRadioListTileWithTitle<T>(title)))
      .containsSemantics(
        label: subtitle == null ? title : '$title\n$subtitle',
        isInMutuallyExclusiveGroup: true,
        hasCheckedState: true, isChecked: expectedIsChecked);
```